### ISSUE TITLE
Unable to add custom languages for user interface in admin panel customization

### ISSUE THREAD

**Issue by memmer57** [strapi/strapi#18929]
### Required System information

- Node.js version: 20.9.0
- NPM version: 9.5.0
- Strapi version: 4.15.4
- Database: SQLite
- Operating system: Linux (Debian)
- Is your project Javascript or Typescript: Javascript

### Describe the bug

I want to add the option for strapi users to change their interface language. From [this](https://docs.strapi.io/dev-docs/admin-panel-customization) doc page, I learned that I can customize admin panel by changing /src/admin/app.js file. This doesn't seem to work properly though. I don't see any languages that I add into my config.locales.

### Steps to reproduce the behavior

1. Create new strapi project

```
npx create-strapi-app@latest my-project --quickstart
```

2. In /src/admin rename app.example.js to app.js with following code:

```js
const config = {
  locales: ["en", "fr", "cs"],
};

const bootstrap = (app) => {
  console.log(app);
};

export default {
  config,
  bootstrap,
};
```

3. Clean build the app and run it in development mode

### Expected behavior

When I go to my profile settings, I should see the option to change my user interface language to either English, French or Czech. Instead, I only see the option to set it to English.

### Screenshots

![image](https://github.com/strapi/strapi/assets/72838098/9b2e02e3-6c6c-479f-859c-e54aaadd7b56)

**Comment by quinhatpy** [comment]
I see the same issue, I can't translation text of the default language or any language for admin panel. Please fix it. Thanks.

**Comment by Sergey-Dyachenko** [comment]
For me, I have the same problem

**Comment by pimpelsang** [comment]
Struggling with this as well

**Comment by yyhdev** [comment]
Even if there is a syntax error in the app.tsx file, the program did not throw any exceptions, indicating that the program did not load the app.tsx file

**Comment by memmer57** [comment]
> Even if there is a syntax error in the app.tsx file, the program did not throw any exceptions, indicating that the program did not load the app.tsx file

Yes, I can relate to this as well. When I tried customizing other stuff in App.js, I wasn't able to notice any changes in my Strapi admin panel.

**Comment by foo-io** [comment]
After updating from 4.14.0 to 4.15.4, interface localization disappeared.

**Comment by joshuaellis** [comment]
This should be fixed in `4.15.5`
